### PR TITLE
Make schema validation errors as warnings

### PR DESCRIPTION
**Pull request by Werni**
Schema validation errors are now warnings for all commands except for
validate. This means validation errors does not prevent viewing a
description but prints the errors as warnings.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 245207dac3eb9c5d04cdf5841a9d82bc2d7d2aa8 into 23f8d0bf3105e4507602b1ff020b7db0258181c3 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged db1e057539ffe692d473453383f87b3f23d437b1 into 23f8d0bf3105e4507602b1ff020b7db0258181c3 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged db1e057539ffe692d473453383f87b3f23d437b1 into 50806d9ec88d48511f31c5efd67d677e13e71072 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 79e6968f85c78a7328dd273865363c0684aba82f into 50806d9ec88d48511f31c5efd67d677e13e71072 )

**Comment by cornelius** [comment]
This doesn't implement that the `upgrade-format` command fails when there are validation errors. As upgrades on an invalid format can cause data loss, we should include that in this patch.

**Comment by cornelius** [comment]
`machinery validate` should not only show the messages from the validation errors but also a header or something like that, so that it's clear if the description validates or not.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 79e6968f85c78a7328dd273865363c0684aba82f into e75bbc9df9cba70819d6ee7e909dbb12878992ca )

File: spec/support/system_description_factory.rb [hunk 1]
```diff
@@ -82,7 +82,11 @@ def create_test_description(options = {})
 
     if options[:json]
       manifest = Manifest.new(name, options[:json])
-      manifest.validate
+      if options[:warnings]
```
Comment (cornelius) [inline spec/support/system_description_factory.rb]: For what is this needed? Where is it used?

File: spec/unit/manifest_spec.rb [hunk 2]
```diff
@@ -92,6 +92,44 @@
         }
       EOT
       expect {
+        manifest.validate!
+      }.not_to raise_error
+    end
+  end
+
+  describe "#validate" do
+    it "shows a warning when validates compatible descriptions" do
```
Comment (cornelius) [inline spec/unit/manifest_spec.rb]: It should either be "when validating" or "when it validates".

File: spec/unit/manifest_spec.rb [hunk 3]
```diff
@@ -92,6 +92,44 @@
...
+      manifest = Manifest.new("name", <<-EOT)
+        {
+          "meta": {
+            "os": "invalid"
+          }
+        }
+      EOT
+      expect {
         manifest.validate
       }.not_to raise_error
```
Comment (cornelius) [inline spec/unit/manifest_spec.rb]: The check is not consistent with the name of the test. If it doesn't valid it should cause an error.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 79e6968f85c78a7328dd273865363c0684aba82f into 505dfe4042ed8d49efe39d4deb5426506d6b4a60 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 79e6968f85c78a7328dd273865363c0684aba82f into d5126d6c5c772df1d54b3c9f95f153c998c35299 )

**Pull request closed without merging**